be/src/storage/metadata_util.cpp

```diff
@@ -256,16 +259,39 @@ Status convert_t_schema_to_pb_schema(const TTabletSchema& tablet_schema, uint32_
 
         has_bf_columns |= column->is_bf_column();
 
-        if (tablet_schema.__isset.indexes) {
-            for (auto& index : tablet_schema.indexes) {
-                if (index.index_type == TIndexType::type::BITMAP) {
-                    DCHECK_EQ(index.columns.size(), 1);
-                    if (boost::iequals(tcolumn.column_name, index.columns[0])) {
-                        column->set_has_bitmap_index(true);
-                        break;
-                    }
+        column_map.insert(std::make_pair(column->name(), column));
+    }
+
+    if (tablet_schema.__isset.indexes) {
+        for (auto& index : tablet_schema.indexes) {
+            TabletIndexPB* index_pb = schema->add_table_indices();
+            index_pb->set_index_id(index.index_id);
+            index_pb->set_index_name(index.index_name);
+            if (index.index_type == TIndexType::type::BITMAP) {
+                DCHECK_EQ(index.columns.size(), 1);
+                if (column_map.count(index.columns[0]) > 0) {
+                    column_map[index.columns[0]]->set_has_bitmap_index(true);
                 }
+                index_pb->set_index_type(IndexType::BITMAP);
+            } else if (index.index_type == TIndexType::type::GIN) {
+                // TODO: Support single column index for now, will support multi-column index in future
+                DCHECK_EQ(index.columns.size(), 1);
+                index_pb->set_index_type(IndexType::GIN);
             }
+
+            for (const auto& index_col_name : index.columns) {
+                CHECK(column_map.count(index_col_name) > 0)
```

Comment:
please don't use CHECK.
This will cause the backend process crashes.
Please return the error instead.